src/solver/nllloss/backward_non_nllloss_4d.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/invoke_params.hpp"
+#include <miopen/nllloss/solvers.hpp>
+
+#include <miopen/nllloss/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/nllloss.hpp>
+#include <miopen/target_properties.hpp>
+#include <miopen/tensor_view.hpp>
+
+#define LOCAL_SIZE_NON_CON_BWD 1024
+
+namespace miopen {
+
+namespace solver {
+
+namespace nllloss {
+
+bool NLLLossUnreduceBackward4d::IsApplicable(
+    const ExecutionContext& context,
+    const miopen::nllloss::UnreduceProblemDescription& problem) const
+{
+    if(problem.GetInputDesc().GetSize() > 4)
```

Comment:
NLLLossUnreduceBackward has 2D, 4D, and ND, so it makes sense to use this kernel only for 4.